src/sql/azdata.proposed.d.ts

```diff
@@ -1741,8 +1741,172 @@ declare module 'azdata' {
+		IgnoreTableOptions: boolean;
+		IgnoreSemicolonBetweenStatements: boolean;
+		IgnoreRouteLifetime: boolean;
+		IgnoreRoleMembership: boolean;
+		IgnoreQuotedIdentifiers: boolean;
+		IgnorePermissions: boolean;
+		IgnorePartitionSchemes: boolean;
+		IgnoreObjectPlacementOnPartitionScheme: boolean;
+		IgnoreNotForReplication: boolean;
+		IgnoreLoginSids: boolean;
+		IgnoreLockHintsOnIndexes: boolean;
+		IgnoreKeywordCasing: boolean;
+		IgnoreIndexPadding: boolean;
+		IgnoreIndexOptions: boolean;
+		IgnoreIncrement: boolean;
+		IgnoreIdentitySeed: boolean;
+		IgnoreUserSettingsObjects: boolean;
+		IgnoreFullTextCatalogFilePath: boolean;
+		IgnoreWhitespace: boolean;
+		IgnoreWithNocheckOnForeignKeys: boolean;
+		VerifyCollationCompatibility: boolean;
+		UnmodifiableObjectWarnings: boolean;
+		TreatVerificationErrorsAsWarnings: boolean;
+		ScriptRefreshModule: boolean;
+		ScriptNewConstraintValidation: boolean;
+		ScriptFileSize: boolean;
+		ScriptDeployStateChecks: boolean;
+		ScriptDatabaseOptions: boolean;
+		ScriptDatabaseCompatibility: boolean;
+		ScriptDatabaseCollation: boolean;
+		RunDeploymentPlanExecutors: boolean;
+		RegisterDataTierApplication: boolean;
+		PopulateFilesOnFileGroups: boolean;
+		NoAlterStatementsToChangeClrTypes: boolean;
+		IncludeTransactionalScripts: boolean;
+		IncludeCompositeObjects: boolean;
+		AllowUnsafeRowLevelSecurityDataMovement: boolean;
+		IgnoreWithNocheckOnCheckConstraints: boolean;
+		IgnoreFillFactor: boolean;
+		IgnoreFileSize: boolean;
+		IgnoreFilegroupPlacement: boolean;
+		DoNotAlterReplicatedObjects: boolean;
+		DoNotAlterChangeDataCaptureObjects: boolean;
+		DisableAndReenableDdlTriggers: boolean;
+		DeployDatabaseInSingleUserMode: boolean;
+		CreateNewDatabase: boolean;
+		CompareUsingTargetCollation: boolean;
+		CommentOutSetVarDeclarations: boolean;
+		BlockWhenDriftDetected: boolean;
+		BlockOnPossibleDataLoss: boolean;
+		BackupDatabaseBeforeChanges: boolean;
+		AllowIncompatiblePlatform: boolean;
+		AllowDropBlockingAssemblies: boolean;
+		DropConstraintsNotInSource: boolean;
+		DropDmlTriggersNotInSource: boolean;
+		DropExtendedPropertiesNotInSource: boolean;
+		DropIndexesNotInSource: boolean;
+		IgnoreFileAndLogFilePath: boolean;
+		IgnoreExtendedProperties: boolean;
+		IgnoreDmlTriggerState: boolean;
+		IgnoreDmlTriggerOrder: boolean;
+		IgnoreDefaultSchema: boolean;
+		IgnoreDdlTriggerState: boolean;
+		IgnoreDdlTriggerOrder: boolean;
+		IgnoreCryptographicProviderFilePath: boolean;
+		VerifyDeployment: boolean;
+		IgnoreComments: boolean;
+		IgnoreColumnCollation: boolean;
+		IgnoreAuthorizer: boolean;
+		IgnoreAnsiNulls: boolean;
+		GenerateSmartDefaults: boolean;
+		DropStatisticsNotInSource: boolean;
+		DropRoleMembersNotInSource: boolean;
+		DropPermissionsNotInSource: boolean;
+		DropObjectsNotInSource: boolean;
+		IgnoreColumnOrder: boolean;
+		DoNotDropObjectTypes: SchemaObjectType[];
+		ExcludeObjectTypes: SchemaObjectType[];
+	}
+
+	export enum SchemaObjectType {
```

Comment:
Same here, can you add a test that fails when a new option is introduced in DacFx, but not present here?

Reply:
same as above